src/python/dxpy/bindings/apollo/expression_matrix_transformation.py

```diff
@@ -0,0 +1,53 @@
+
+def expression_transform(results_list):
+    """
+    results_list: list of dictionaries, each of the form
+    {
+        "feature_id":<feature_id>,
+        "sample_id":<sample_id>,
+        "expression":<expression>
+    }
+    Additional key:value pairs are possible, but are not included in the transformed output
+
+    How it works:
+    First, create a dictionary where each key is a sample id, and the values are the feature_id:expression pairs
+    Then, convert it to a list of dictionaries (the format that output handling is expecting)
+    Each item in the list is a dictionary containing one "sample_id":sample_id pair, and
+    multiple feature_id:expression pairs
+
+    Returns:
+    A list of dictionaries, each of the form:
+    {
+        "sample_id":<sample_id>,
+        <feature_id_1>:<expression_1>
+        <feature_id_2>:<expression_2>
+        ...
+        <feature_id_n>:<expression_n>
+    }
+    """
+    transformed_dict = {}
+    # create a dict of the form
+    # {
+    #   <sample_id>:{
+    #       "sample_id":<sample_id>,
+    #       <feature_id_1>:<expression_1>,
+    #       <feature_id_2>:<expression_2>,
+    #        etc.
+    #   }
+    # 
+    # }
+    colnames = set()
```

Comment:
How about collecting all colnames already here?

```suggestion
    f_ids = set(row["feature_id"] for row in results_list)
    colnames=["sample_id"]
    colnames.extend(list(f_ids))
```